app/models/concerns/bulkrax/has_matchers.rb

```diff
@@ -103,12 +113,21 @@ def multiple_metadata?(name, node_name, node_content, object = false)
       end
     end
 
-    def matched_metadata?(matcher, multiple, name, node_content, object = false)
+    def matched_metadata?(matcher, multiple, name, node_content, object, object_multiple)
       result = matcher.result(self, node_content)
       return unless result
 
       if object
-        if multiple
+        if object_multiple
+          index = parsed_metadata[object].find_index { |obj| obj[name].nil? }
+
+          if index.nil?
+            parsed_metadata[object] << {}
+            parsed_metadata[object][parsed_metadata[object].length - 1][name] = Array.wrap(result).join('; ')
+          else
+            parsed_metadata[object][index][name] = Array.wrap(result).join('; ')
+          end
```

Comment:
Please add a comment / comments explaining what these lines are doing and why we're doing it the way we are. Personally, I would not know what this was doing just by looking at the code

Reply:
```suggestion
        if object_multiple
          # find the index of the first object in the `object` array where the `name` key doesn't already exist
          index = parsed_metadata[object].find_index { |obj| obj[name].nil? }

          if index.nil?
            # if all existing objects have our `name` key in it already
            # push a new object to the end of the array and add the `result` to it
            parsed_metadata[object] << {}
            parsed_metadata[object][parsed_metadata[object].length - 1][name] = Array.wrap(result).join('; ')
          else
            # if an object already exists that doesn't have our `name` key in it, add the `result` to that object
            parsed_metadata[object][index][name] = Array.wrap(result).join('; ')
          end
```